store/tikv/2pc_fail_test.go

```diff
@@ -144,3 +146,30 @@ func (s *testCommitterSuite) TestFailPrewriteRegionError(c *C) {
 	err = committer.prewriteMutations(NewBackofferWithVars(ctx, 1000, nil), committer.mutations)
 	c.Assert(err, NotNil)
 }
+
+// TestFailCommitPrimaryRpcErrors tests rpc errors are handled properly when
+// committing primary region task.
+func (s *testCommitterSuite) TestFailAsyncCommitPrewriteRpcErrors(c *C) {
```

Comment:
Do we need to add a case that a transation from client `A` fails to get the prewrite response and actually it succeeds, another transaction from client `B` recoveries the transaction, and query again from client `A` the result is obtained.